service/history/decisionTaskHandler.go

```diff
@@ -206,14 +206,23 @@ func (handler *decisionTaskHandlerImpl) handleDecisionScheduleActivity(
 		return nil
 	}
 
-	scheduleEvent, _ := handler.mutableState.AddActivityTaskScheduledEvent(handler.decisionTaskCompletedID, attr)
-	handler.transferTasks = append(handler.transferTasks, &persistence.ActivityTask{
-		DomainID:   targetDomainID,
-		TaskList:   attr.TaskList.GetName(),
-		ScheduleID: scheduleEvent.GetEventId(),
-	})
-
-	return nil
+	scheduleEvent, _, err := handler.mutableState.AddActivityTaskScheduledEvent(handler.decisionTaskCompletedID, attr)
+	switch err.(type) {
+	case nil:
```

Comment:
not really, just want to structure the code in a way that contains less if / else

if this is not handled like above, it will looks like this
```
if err != nil {
  if _, ok := err.(*workflow.BadRequestError); ok {
    // some logic
    return nil / err
  }
  return err
}

// some other logic
```